sphinx/writers/latex.py

```diff
@@ -1738,7 +1738,8 @@ def depart_footnotemark(self, node: Element) -> None:
     def visit_footnotetext(self, node: Element) -> None:
         label = cast(nodes.label, node[0])
         self.body.append('%%\n\\begin{footnotetext}[%s]'
-                         '\\sphinxAtStartFootnote\n' % label.astext())
+                         '\\phantomsection\\label{fn:%s}%%\n'
```

Comment:
as ``%s`` is a number, I think ``fn:%s`` car never duplicate another label. However in case of doubt, we could simply add a space before ``fn``. And adjust the same in the latex code. Because never will Sphinx otherwise generate space prefixed labels, and a space prefixed label is not a problem for LaTeX.

Reply:
I noticed docutils supports symbol-footnotes feature. It assigns non-numbers (ex. asterisk, daggar, etc.) to the footnotes. And it is not supported in our current implementation.
https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#auto-symbol-footnotes

Additionally, we have an idea for the scope of footnote numbers (per chapter, per section, per page and so on). The duplicated footnote number will be shown if we'll support it in the future.

Therefore, it might be better to assign node_id to the footnote definitions and use it on footnote references.

Reply:
In standard LaTeX (this is not modified by hyperref or footnotehyper) whenever ``[..]`` syntax is used with ``\footnote`` (or ``footnote`` environment with footnotehyper), ``\footnotemark``, ``\footnotetext`` (or rather ``footnotetext`` for us with footnotehyper), it is mandatorily a number, as it will be assigned to a TeX counter locally, either the footnote counter or the mpfootnote counter. This is done because LaTeX then uses one of the two macros `\thefootnote` or `\thempfootnote` which convert the counter into some printed form which can be customized (using for example roman numerals i, ii, iii, ...).

LaTeX has also a way to use symbols, and for this `\thefootnote` is defined to use `\@fnsymbol` which is a converter from numbers to symbols. The canonical sequence has only 9 symbols, starting at number 1 and gives  ∗ † ‡ § ¶ ‖ ∗∗ †† ‡‡. Probably there is some package to modify this hardcoded choice, or I can do the coding. But whatever happens, we can not use ``[...]`` syntax with something else.

Writing this, I have an idea, that we could use ``[...]`` with negative numbers for example ``[-1]``. Then we only have to modify ``\thefootnote``  to test if input is negative and in that case use a predefined symbol list. 

The other question you raise is the scope of footnote numbering.

- When using `\begin{footnote}[Number]`, anyhow hyperref uses an internal counter `Hfootnote` to label the anchor so there is not much problem: we keep memory of this anchor label in a macro indexed by Number (to allow `\sphinxfootnotemark[Number]` later to refer same footnote), so when scope of Sphinx footnote numbering is reset, the macros will start getting redefined by new footnotes with same number. Problem is that `\sphinxfootnotemark` first checks if the storage macro is defined, so we have to make them undefined at start of scope else there will be confusion with `\sphinxfootnotemark/footnotetext` usage. I will think about it. (I thougt about it by end of next item...)
- When using `\sphinxfootnotemark[N]` and `\begin{footnotetext}[N]` we can solve the scope problem by letting the `fn:` prefix be dynamical with the scope. It could be internally `\spx@fnprefix` and redefined at each scope reset via `\sphinxsetfnprefix{<some new unique prefix>}` from writers/latex.py output. In fact we could even simply have mark-up `\sphinxstepscope`, we would have a counter, at each scope change, it is incremented, the we use as prefix the value of the counter followed by some separator like `:`. If nothing else in Sphinx can generate `Number1:Number2` label then we are fine.

The idea of the scope counter can also be used to address issue in my first item, in place of making undefined all macros holding the hyperref internal anchor labels (because my code in `\sphinxfootref` tests is defined or not) at each scope change, we shall simply integrate the scope number into the storage macro names themselves so no duplicates can ever arise. Ok, this will work.

So I will push a modification to make this PR "scope"-ready. (memo:  `\sphinxstepscope` will have to be inserted before `\section` in future, in case a footnote is inside title).

As per making it "symbol"-ready, as I said, anyhow the LaTeX syntax requires `[..]` to contain always a number, so we will not put the symbol in there, but we can put a negative number and then have some LaTeX internals test sign of number and if negative use a symbol associated to it according to some dispatch macro (LaTeX's default one is `\@fnsymbol` and has only 9 predefined symbols but we can code ourselves the DocUtils specification; notice however that if symbols are Unicode we will have to let them be known to pdflatex, find suitable font holding them, and even with lualatex/xelatex, we have to be sure our font holds them, so there will be complications). Anyway symbols will never make it into labels per se, only the negative numbers which will serve to index them.

Reply:
> Additionally, we have an idea for the scope of footnote numbers (per chapter, per section, per page and so on). The duplicated footnote number will be shown if we'll support it in the future.

My idea here is fine for per chapter, per section, and to implement it we only need now to add `\stepcounter{sphinxscope}` at each scope change which will reset our generated footnote numbers which make it hard-coded into the latex file.

Per page requires a complete refactoring of our handling of footnote numbering, because only during pdf build will the page numbers be discovered. Currently this looks like difficult task. I think some improved upstream footnotehyper could help which will work around difficulties of LaTeX core since 30years with using multiple times  `\footnotemark` in a table for example and `footnotetext` after table (without even mentioning the added hyperlink complications). This is one of the reasons why in the past we opted for numbering footonotes via `[..]` output latex mark-up.

For per page, we could keep the same output actually (as the LaTeX code can choose to basically ignore it) but we need then some footnotehyper modifications. I don't quite see all problems in advance here, which will arise, and it will require some effort, it is not obvious task.

Reply:
> I noticed docutils supports symbol-footnotes feature. It assigns non-numbers (ex. asterisk, daggar, etc.) to the footnotes. And it is not supported in our current implementation.
> https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#auto-symbol-footnotes

I looked at it, and it is feasible in latex with same default symbols (the last four are available via math mode ``$\spadesuit$`` etc...). Once we support symbol-footnotes at core Sphinx, we can implement them in latex via using negative numbers for ``footnote, footnotetext, \sphinxfootnotemark``: ``[-1]`` to ``[-10]`` for the ten symbols, ``[-11]``, ``[-12]`` ... will mean to double the symbol, ``[-21]``, ..., ``[-30]``  will triple the symbol. It is all feasible. (we can only use ``[..]`` syntax with numbers, it is a core latex requirement).